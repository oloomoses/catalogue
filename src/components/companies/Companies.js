import React from 'react';
import { useSelector } from 'react-redux';
import { Row } from 'react-bootstrap';
import Spinner from '../layout/Spinner';
import StockItems from '../stock/StockItems';

const Companies = () => {
  const loading = useSelector((state) => state.loading);
  const companies = useSelector((state) => state.companies);
  if (loading) {
    return <Spinner />;
  }
  return (
    <div>
      <Row>
        {companies.map((company) => (
          <StockItems key={company.ticker} company={company} />
        ))}
      </Row>
    </div>
  );
};

export default Companies;
