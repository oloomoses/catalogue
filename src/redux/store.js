import { createStore } from 'redux';
import stockReducer from './stock/stockReducer';

const store = createStore(stockReducer);

export default store;
