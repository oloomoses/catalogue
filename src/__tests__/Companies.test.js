import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Companies from '../components/companies/Companies';
import store from '../redux/store';

describe('renders correctly', () => {
  it('Contains the StockItems component', () => {
    const wrapper = shallow(<Provider store={store}><Companies /></Provider>);
    expect(wrapper.find('StockItems')).toBeTruthy();
  });

  it('Contains the Spinner component', () => {
    const wrapper = shallow(<Provider store={store}><Companies /></Provider>);
    expect(wrapper.find('Spinner')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<Provider store={store}><Companies /></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
