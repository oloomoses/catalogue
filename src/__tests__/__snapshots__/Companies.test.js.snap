// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  />
</div>
`;
